include/sys/range_tree.h

```diff
@@ -48,6 +48,31 @@ typedef enum zfs_range_seg_type {
 	ZFS_RANGE_SEG_NUM_TYPES,
 } zfs_range_seg_type_t;
 
+/*
+ * Use case flags to support the zfs_recover_rt parameter.
+ *
+ * The range tree's logic needs to know the context in order to correctly
+ * recover from an unexpected situation by exchanging potential data loss for
+ * a potential space leak:
+ *
+ * - If it knows that the tree represents allocated space then it should better
+ *   perform an unexpected addition to the tree.
+ *
+ * - Similarly, if it's about free space (aka allocatable) then it should
+ *   perform unexpected removals instead of silently ignoring the issue.
+ *
+ * An unknown case means to ignore unexpected additions/removals as a recovery
+ * measure.
+ *
+ * In any case, unexpected actions are logged with extra details such as
+ * a range tree instance string, which can be marked as dynamic to be freed
+ * along with the tree instance destruction.
+ */
+#define	ZFS_RANGE_TREE_UC_UNKNOWN		(1 << 0)
+#define	ZFS_RANGE_TREE_UC_ALLOCATED_SPACE	(1 << 1)
+#define	ZFS_RANGE_TREE_UC_FREE_SPACE		(1 << 2)
+#define	ZFS_RANGE_TREE_UC_DYN_INSTANCE		(1 << 16)
```

Comment:
DYN_INSTANCE is not a use case.  It is more of a general flag.

Reply:
Yeah, initially it was only about use cases, that's why not a use-case one was added with `<< 16` to differentiate a little. Now, having agreed on a common `flags` concept it's time to rename them to `ZFS_RANGE_TREE_F_*` where `UC` is just a sub-set of the flags.